### PR TITLE
Add ProbeOutput for points defined on a line

### PR DESCRIPTION
**Pull request by angranl-flex**

### CODE REVIEW

File: flow360/component/simulation/translator/solver_translator.py [hunk 1]
```diff
@@ -227,14 +228,28 @@ def inject_isosurface_info(entity: Isosurface):
...
     return {
         "monitorLocations": [item.location.value.tolist() for item in entity.stored_entities],
         "type": "probe",
     }
 
 
-def inject_surface_robe_info(entity: EntityList):
+def inject_surface_probe_info(entity: EntityList):
     """inject entity info"""
+    if isinstance(entity.stored_entities[0], PointArray):
```
Comment (benflexcompute) [inline flow360/component/simulation/translator/solver_translator.py:246]: Can we add validation that there is only 1 type of entity in entity list?
BTW can you remind me why we decided this way? Why not append isolated points to the line?

Comment (angranl-flex) [reply flow360/component/simulation/translator/solver_translator.py:246]: Yes, the validation is added

File: flow360/component/simulation/outputs/outputs.py [hunk 2]
```diff
@@ -152,14 +153,27 @@ class ProbeOutput(Flow360BaseModel):
...
         """Load probe point locations from a file."""
         raise NotImplementedError("Not implemented yet.")
 
+    @pd.field_validator("entities", mode="after")
+    @classmethod
+    def check_entities_type(cls, value):
+        """check to ensure every entity has the same type"""
+        if value is None:
+            return value
+        for entity in value.stored_entities:
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:170]: Can you encapsulate this into a function and call it in both classes instead?

Comment (benflexcompute) [reply flow360/component/simulation/outputs/outputs.py:170]: You can put it in Flow360/flow360/component/simulation/validation/validation_output.py

Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Updated.

File: flow360/component/simulation/outputs/outputs.py [hunk 3]
```diff
@@ -152,14 +156,20 @@ class ProbeOutput(Flow360BaseModel):
...
     output_fields: UniqueItemList[CommonFieldNames] = pd.Field()
     output_type: Literal["ProbeOutput"] = pd.Field("ProbeOutput", frozen=True)
 
     def load_point_location_from_file(self, file_path: str):
         """Load probe point locations from a file."""
         raise NotImplementedError("Not implemented yet.")
 
+    @pd.field_validator("entities", mode="after")
+    @classmethod
+    def check_entities_type(cls, value):
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:169]:
```suggestion
    def check_unique_probe_type(cls, value):
```
same for the `_check_entities_type`

File: flow360/component/simulation/outputs/outputs.py [hunk 4]
```diff
@@ -168,13 +178,19 @@ class SurfaceProbeOutput(Flow360BaseModel):
...
+    entities: EntityList[Point, PointArray] = pd.Field(None, alias="probe_points")
     # Maybe add preprocess for this and by default add all Surfaces?
     target_surfaces: EntityList[Surface] = pd.Field()
 
     output_fields: UniqueItemList[SurfaceFieldNames] = pd.Field()
     output_type: Literal["SurfaceProbeOutput"] = pd.Field("SurfaceProbeOutput", frozen=True)
 
+    @pd.field_validator("entities", mode="after")
+    @classmethod
+    def check_entities_type(cls, value):
```
Comment (benflexcompute) [inline flow360/component/simulation/outputs/outputs.py:190]: same here. Let's try making the name as explanatory as possible.

Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Updated

File: flow360/component/simulation/outputs/outputs.py (hunk 3 above)
Comment (angranl-flex) [reply flow360/component/simulation/outputs/outputs.py]: Done

**Review by benflexcompute — APPROVED** [review]

**Pull request merged**